### PR TITLE
Fixes #6196: properly check if ldap is up and running to avoid skipping ...

### PR DESCRIPTION
**Pull request by ncharles**
...ldap upgrade scripts

### CODE REVIEW

**Comment by ncharles** [comment]
@peckpeck @Kegeruneku : could you both have a look at that ?

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 1]
```diff
@@ -292,16 +290,20 @@ if [ -d ${CONFIGURATION_REPOSITORY}/.git ];then
 fi
 
 # Check the LDAP and PostgreSQL daemons status before upgrading
-if [ ${LDAP_EXISTS} -ne 0 ]; then
-  echo -n "INFO: Checking LDAP service status..."
-  retry_wrapper "${LDAPSEARCH} -s one -b 'cn=rudder-configuration' cn >/dev/null 2>&1" "LDAP"
-  echo " OK"
-fi
+
+# Using retry_wrapper is necessary because during an upgrade, slapd may be restarted, so we need to wait
```
Comment (jooooooon) [inline rudder-webapp/SOURCES/rudder-upgrade]: I suggest using this instead, which is more generic:

```
${LDAPSEARCH} -x -b "" -s base subschemaSubentry
```

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 2]
```diff
@@ -292,16 +290,20 @@ if [ -d ${CONFIGURATION_REPOSITORY}/.git ];then
...
+# for it to be available again
+echo -n "INFO: Checking LDAP service status..."
+retry_wrapper "${LDAPSEARCH} -b cn=Subschema -s base '(objectClass=subschema)' dn >/dev/null 2>&1" "LDAP"
```
Comment (jooooooon) [inline rudder-webapp/SOURCES/rudder-upgrade]: I suggest using this instead, which is more generic:

```
${LDAPSEARCH} -x -b "" -s base subschemaSubentry
```

**Comment by peckpeck** [comment]
Good !

**Pull request merged**